api/kyverno/v1/clustercleanuppolic_types.go

```diff
@@ -0,0 +1,47 @@
+package v1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +kubebuilder:object:root=true
+// +kubebuilder:subresource:status
+// +kubebuilder:printcolumn:name="Schedule",type=string,JSONPath=".spec.schedule"
+// +kubebuilder:printcolumn:name="Age",type="date",JSONPath=".metadata.creationTimestamp"
+
+// CleanupPolicy declares cleanup behaviors for matching resources.
+type ClusterCleanupPolicy struct {
+	metav1.TypeMeta   `json:",inline,omitempty" yaml:",inline,omitempty"`
+	metav1.ObjectMeta `json:"metadata,omitempty" yaml:"metadata,omitempty"`
+
+	// Spec declares policy behaviors.
+	Spec ClusterCleanupPolicySpec `json:"spec" yaml:"spec"`
+
+	// Status contains policy runtime data.
+	// +optional
+	Status ClusterCleanupPolicyStatus `json:"status,omitempty" yaml:"status,omitempty"`
+}
+
+type ClusterCleanupPolicySpec struct {
+	// The schedule in Cron format
+	// +optional
+	Schedule string `json:"schedule,omitempty" yaml:"schedule,omitempty"`
+
+	// Conditions defines conditions used to select resources which user needs to delete
+	// +optional
+	Conditions *AnyAllConditions `json:"conditions,omitempty" yaml:"conditions,omitempty"`
+}
+
+type ClusterCleanupPolicyStatus struct {
+	Status []metav1.Condition `json:"conditions,omitempty" patchStrategy:"merge" patchMergeKey:"type" protobuf:"bytes,1,rep,name=conditions"`
```

Comment:
Why is this field called `Status` ?

Reply:
I thought we can add/update the status for the CR in this field. Based on this [format](https://pkg.go.dev/k8s.io/apimachinery/pkg/apis/meta/v1#Condition).

Reply:
But why is the field called `Status` ? Wouldn't `Conditions` be more appropriate ?